Fixes #4- Removed the exposed API Keys in Code and ensured security

vaibhav-katyal opened the pull request:
## Description
This PR addresses **issue #4** by removing hardcoded/exposed API keys from the codebase and implementing secure handling practices.

## Changes Made
- Removed all exposed API keys from source files  
- Updated code to fetch API keys securely from environment variables or a configuration file

## Impact
- Enhances overall project security  
- Prevents unauthorized access and key misuse  
- Aligns with best practices for credential management

vaibhav-katyal committed:
Fixes #4- Removed the exposed API Keys in Code and ensured security

150202-Pratham commented:
Hi vaibhav i would request you to kindly first check and review whether the data is flowing and the news is properly visible on the UI as i think it will give error of not fetching the data

vaibhav-katyal commented:
But Pratham, that is why you asked the user to create a .env file and add their own credentials if they wanted the project to work. 
As in ReadMe file 👇
<img width="1087" height="210" alt="image" src="https://github.com/user-attachments/assets/b29a816d-ce5c-40da-beeb-7784e94fc51c" />


However, you left the API key exposed in the JavaScript files by importing it from .env and also hardcoding it after the || operator as a fallback. This is not a good practice — API keys should never be exposed in the source code. The hardcoded key can be used only for local testing purposes, not in production.

If you want to demonstrate the full working of the project to users, you can first deploy the complete version and then add your own API Key or the environment variables securely within the deployment platform’s settings instead of keeping them in your code.

Thats why i have removed the fallback one.

150202-Pratham commented:
Hi, I knew that i have Exposed the Api Key Directly there after the '||' operator but did you verfied that without using this ' || ' operator the data is visible onto your UI.

if so , then i request you to kindly make a new Branch and then raise a new Commit to it so that it becomes easy for me to review your changes before merging directly to master branch

vaibhav-katyal committed:
Made the environment variables according to the vite environment

vaibhav-katyal commented:
I found the issue that caused the app to stop working when the fallback hardcoded API key was removed. The problem was with the environment variable names — they were defined using the **React** naming convention (`REACT_APP_...`), but this project uses **Vite**, which follows a different convention.

#### Changes Made:
- Renamed environment variables:
  - `REACT_APP_NEWS_API_KEY` → `VITE_NEWS_API_KEY`
  - `REACT_APP_WEATHER_API_KEY` → `VITE_WEATHER_API_KEY`
- Updated all references in the source files.
- Updated the **README.md** file to guide users in setting up their environment correctly.

This ensures the app can now properly fetch API keys from the environment variables in a Vite setup.

vaibhav-katyal commented:
If the project is now working correctly after these changes, please **merge my PR** and label it **`hacktoberfest-accepted`**.

Currently, this repository is showing as **"Not participating"** on [hacktoberfest.com](https://hacktoberfest.com).  

<img width="1484" height="303" alt="image" src="https://github.com/user-attachments/assets/8e99eb39-f6fa-4961-99fa-3ae394e74c1d" />

Merging the PR with the `hacktoberfest-accepted` label will ensure it counts toward the event.

Thank you! 😊

150202-Pratham reviewed (approved):
Hello Vaibhav that was a great work done by you;
